src/main/java/io/mosip/digitalcard/service/impl/PDFCardServiceImpl.java

```diff
@@ -358,5 +360,9 @@ private byte[] generateUinCard(InputStream in, String password) {
 
 		return pdfSignatured;
 	}
+
+	public void loginErrorDetails(String rid, String errorMsg){
+		digitalCardTransactionRepository.updateErrorTransactionDetails(rid,"ERROR",errorMsg,LocalDateTime.now(),Utility.getUser());
```

Comment:
This is the layer in which they will have their customizations, they cannot interact with DB, we have to throw exception and in the digital card service impl, catch this and then update the db

Reply:
taken care